### PR TITLE
feat: basic auth

### PR DESCRIPTION
**Pull request by Libertonius**

### CODE REVIEW

File: config/apolloClient.ts [hunk 1]
```diff
@@ -1,41 +1,131 @@
...
 import { getMainDefinition } from "@apollo/client/utilities";
 import { GraphQLWsLink } from "@apollo/client/link/subscriptions";
+import { TokenRefreshLink } from "apollo-link-token-refresh";
 import { createClient } from "graphql-ws";
+import decodeJWT from "jwt-decode";
+
+import { fetchRefreshToken } from "@/hooks/useAuth";
+import AuthTokensStore from "@/stores/AuthTokensStore";
+
+type headers = {
```
Comment (DefK1lla) [inline config/apolloClient.ts:16]: let's write type names in PascalCase

File: config/apolloClient.ts [hunk 2]
```diff
@@ -1,41 +1,131 @@
...
-  httpLink
-);
+const getHeaders = () => {
+  const headers = {} as headers;
+  const { accessToken } = AuthTokensStore.getState();
+  if (accessToken) headers.Authorization = `Bearer ${accessToken}`;
+  return headers;
+};
+
+const operationIsSubscription = (operation: any): boolean => {
```
Comment (DefK1lla) [inline config/apolloClient.ts:30]: let's not use the "any" type

File: src/hooks/useAuth.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        "Content-Type": "application/json",
+      },
+    }
+  );
+};
+
+export default () => {
+  const authTokensState = AuthTokensStore.getState();
+
+  const login = async (values: any) => {
```
Comment (DefK1lla) [inline src/hooks/useAuth.ts:48]: let's not use the "any" type

File: src/components/SignUpForm/index.tsx [hunk 4]
```diff
@@ -48,16 +48,16 @@ const SignUpForm: FC<SignUpProps> = ({ onSubmit }) => {
         <Text>{t("common:words.or")}</Text>
       </div>
 
-      <Form className={styles.form}>
+      <Form className={styles.form} onFinish={handleSubmit}>
```
Comment (DefK1lla) [inline src/components/SignUpForm/index.tsx:51]: prop onfinish is not needed, the handler is already on the button. in addition, the handleSubmit function expects a callback to be passed to it

**Review by DefK1lla — CHANGES_REQUESTED** [review]

File: config/apolloClient.ts (hunk 2 above)
Comment (ifokeev) [reply config/apolloClient.ts:30]: @DefK1lla how to change?
Please provide solution

File: config/apolloClient.ts (hunk 1 above)
Comment (ifokeev) [reply config/apolloClient.ts:16]: so should be `Headers`

File: config/apolloClient.ts (hunk 2 above)
Comment (DefK1lla) [reply config/apolloClient.ts:30]: @ifokeev inside apollo there is an interface called Operation. 
image
image

**Review by ifokeev — APPROVED** [review]

**Pull request merged**